### DISCUSSION TITLE
Cross compile `aarch64` linux target on `x64` linux host?

### DISCUSSION THREAD

**Discussion by Zingam** [microsoft/vcpkg#39492] [Q&A]
I want to compile `curl` on x64 linux to a RPI aarch64 target using this GCC 14 toolchain: 

https://github.com/tttapa/docker-arm-cross-toolchain

By default vcpkg picks up the x64-linux triplet. If I set: "VCPKG_TARGET_TRIPLET": "arm64-linux" a different kind of errors.

Is it possible to use vcpkg when cross compiling to a different architecture?

**Comment by dg0yt** [comment]
Yes.

**Comment by Zingam** [comment]
Ausgezeichnet! Are there any examples how to properly setup a cross-compilation with gcc for arm or even RPI?
I've followed the tutorials by the author of the above tool chain. I can Compile a statically linked executable but when I included vcpkg to Compile non-header only library, it appears that vcpkg is looking for Ninja (capital letter) and a different compiler than the one I have provided via cmake presets.
I do have ninja installed.

**Reply by dg0yt** [reply]
I do not use any specific setup. I simply use the cross compilers etc. from Ubuntu. It just "works for me".

Ninja really isn't the tool which will make problems because this the same binary as for host builds. 

Customizations might be needed for the compiler etc. if they don't match the expected names. Customization points are variables in (custom) triplets (for script mode) and the chainloaded cmake toolchain (for project mode).

You can make proper reproducible reports if you want detailed answers. It is not even clear if you use an outdated vcpkg.

**Reply by Zingam** [reply]
Thanks. That's a good point. I've been looking for general directions and examples how to setup a project targeting Raspberry PI on WSL2. I am interested in using the latest GCC or Clang (or both).
As seen here. vcpkg doesn't support aarch64 oficially:
https://devblogs.microsoft.com/cppblog/whats-new-in-vcpkg-may-2024-dedicated-pages-for-each-package-on-vcpkg-io/
 I followed loosely this guide and used the toolchain:
- https://tttapa.github.io/Pages/Raspberry-Pi/C++-Development-RPiOS/index.html
- https://github.com/tttapa/docker-arm-cross-toolchain
So I have setup a small project using CMake + CMake presets + Visual Studio Code. I added the `vcpkg` toolchain and it seems to compile a header only libraries well. 

When I set `  "VCPKG_TARGET_TRIPLET": "arm64-linux"` I get strange errors:

> CMake Error at CMakeLists.txt:11 (enable_language):
>  The CMAKE_C_COMPILER:
>
>    aarch64-linux-gnu-gcc
>
>  is not a full path and was not found in the PATH.


And also:

> [cmake]   vcpkg install failed.  See logs for more information:
[cmake]   /home/hristo/Projects/CMakeTest/out/build/RaspberryPI-debug/vcpkg-manifest-install.log
[cmake] Call Stack (most recent call first):
[cmake]   /snap/cmake/1401/share/cmake-3.30/Modules/CMakeDetermineSystem.cmake:146 (include)
[cmake]   CMakeLists.txt:8 (project)
[cmake] 
[cmake] 
[cmake] CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.


[CMakeTest.zip](https://github.com/user-attachments/files/16010384/CMakeTest.zip)
